cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -1011,3 +1030,16 @@ func isMachineFailedOrTerminating(machine *v1alpha1.Machine) bool {
 	}
 	return false
 }
+
+func filterExtendedResources(allResources v1.ResourceList) (extendedResources v1.ResourceList) {
+	extendedResources = make(v1.ResourceList)
+	for _, n := range knownResourceNames {
```

Comment:
Aren't we just filtering the `knownResourceNames` in this? What if I have some other extended resource which is not part of this list? I think in general that would be the case

Reply:
We are  the removing the `knownResources` and keeping all other extended resources in this function.  Will add a  doc and unit-test to make this clear.